server/actions/elections.go

```diff
@@ -12,51 +12,80 @@ import (
 	uuid "github.com/satori/go.uuid"
 )
 
+// CreateElection creates an election with the given name.
+// All other fields are set to their defaults and has to be changed
+// using the edit endpoint.
+// Default values:
+// - Description: ""
+// - OpenTime, CloseTime: null
+// - Published, Finalized: false
 func CreateElection(c *gin.Context) {
 	body := struct {
 		Name string `json:"name" binding:"required"`
 	}{}
 
-	if !util.TryParseJsonBody(c, &body) {
+	if err := c.BindJSON(&body); err != nil {
+		fmt.Println(err)
+		c.String(http.StatusBadRequest, util.BadParameters)
 		return
 	}
 
-	fmt.Println(body)
-
 	db := database.GetDB()
 	defer database.ReleaseDB()
-	db.Create(&database.Election{
+	election := database.Election{
 		ID:          uuid.NewV4(),
 		Name:        body.Name,
 		Description: "",
 		OpenTime:    sql.NullTime{Valid: false},
 		CloseTime:   sql.NullTime{Valid: false},
 		Published:   false,
 		Finalized:   false,
-	})
+	}
+	if err := db.Create(&election).Error; err != nil {
+		fmt.Println(err)
+		c.String(http.StatusBadRequest, util.RequestFailed)
+		return
+	}
 
-	c.String(http.StatusOK, "200 OK")
+	c.JSON(http.StatusOK, election)
 }
 
+// EditElection updates specific fields for the specified election.
+// Individual fields can be skipped in the request body. All skipped fields
+// will not be affected in the database.
+// Allowed fields are: Name, Description, OpenTime, CloseTime
 func EditElection(c *gin.Context) {
-	electionId, success := util.TryParseUuidInPath(c)
 	body := struct {
 		Name        *string        `json:"name"`
 		Description *string        `json:"description"`
 		OpenTime    *util.NullTime `json:"openTime"`
 		CloseTime   *util.NullTime `json:"closeTime"`
 	}{}
+	electionId, err := uuid.FromString(c.Param("id"))
 
-	if !success || !util.TryParseJsonBody(c, &body) {
+	if err != nil {
+		fmt.Println(err)
+		c.String(http.StatusBadRequest, util.BadUUID)
+		return
+	}
+	if err := c.BindJSON(&body); err != nil {
+		fmt.Println(err)
+		c.String(http.StatusBadRequest, util.BadParameters)
 		return
 	}
 
 	election := database.Election{ID: electionId}
 	db := database.GetDB()
 	defer database.ReleaseDB()
-	if err := db.First(&election).Error; err != nil {
+	if err := db.Preload("Candidates").First(&election).Error; err != nil {
 		fmt.Println(err)
-		c.String(http.StatusBadRequest, "invalid election specified")
+		c.String(http.StatusBadRequest, util.InvalidElection)
+		return
+	}
+
+	if election.Finalized {
+		fmt.Println("User tried to edit election after it had been finalized")
+		c.String(http.StatusMethodNotAllowed, "Can't edit finalized election")
```

Comment:
BadRequest? Forbidden?

`The HyperText Transfer Protocol (HTTP) 405 Method Not Allowed response status code indicates that the server knows the request method, but the target resource doesn't support this method.`

Tolkar detta som att denna typ av resurs inte stödjer denna metod (någonsin).